src/lib/types.ts

```diff
@@ -34,3 +38,11 @@ export interface ChannelData {
     name: string;
     private: boolean;
 }
+
+export interface CommentData {
+    user: User; // may need to be User
+    content: string;
+    downvotes: number;
+    upvotes: number;
+    children?: CommentData[];
```

Comment:
Why is children optional?  In theory, an empty array would suffice when there are no children.